Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -1992,6 +1999,14 @@ void MainWindow::ShowAchievementSettings()
   ShowAchievementsWindow();
   m_achievements_window->ForceSettingsTab();
 }
+
+void MainWindow::ToggleHardcore()
```

Comment:
This would be better named `OnHardcoreChanged`. Right now it sounds like calling this function changes the value of `Config::RA_HARDCORE_ENABLED`.

Reply:
Now that I look closer, this function is more or less doing the same thing as AchievementsSettingsWidget::UpdateHardcoreMode. You should set up a config callback that calls that instead of creating a new function here.